src/cxx.c

```diff
@@ -42,6 +42,39 @@ char *libdemangle_handler_cxx(const char *str) {
 			*glibcxx = '\0';
 		}
 	}
+
+	uint32_t len = strlen(p);
+	if (len > 4 && !strncmp(p + len - 4, "_ptr", 4)) {
```

Comment:
What about converting 4 to strlen("_ptr") there?

Reply:
doesn't change much the code.